const a = (s, e, t) => {
  const r = new Error(s);
  r.name = "FetcherError", r.url = e.url, r.status = e.status, r.statusText = e.statusText, r.responseText = t;
  try {
    const c = JSON.parse(t);
    r.responseJson = c;
  } catch {
  }
  return r;
}, o = async (s, e) => {
  const t = await fetch(s, e), r = await t.text();
  if (!t.ok)
    throw a("An error occurred while fetching", t, r);
  try {
    return JSON.parse(r);
  } catch {
    throw a("An error occurred while parsing", t, r);
  }
};
export {
  o as fetchData,
  a as fetcherError
};
